hotel_reservation/models/hotel_reservation.py

```diff
@@ -0,0 +1,1203 @@
+# See LICENSE file for full copyright and licensing details.
+
+import time
+from datetime import datetime, timedelta
+
+import pytz
+from dateutil.relativedelta import relativedelta
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT as dt
+
+
+class HotelFolio(models.Model):
+
+    _inherit = "hotel.folio"
+    _order = "reservation_id desc"
+
+    reservation_id = fields.Many2one(
+        "hotel.reservation", "Reservation", ondelete="restrict"
+    )
+
+    def write(self, vals):
+        context = dict(self._context)
+        context.update({"from_reservation": True})
+        res = super(HotelFolio, self).write(vals)
+        reservation_line_obj = self.env["hotel.room.reservation.line"]
+        for folio in self:
+            reservations = reservation_line_obj.search(
+                [("reservation_id", "=", folio.reservation_id.id)]
+            )
+            if len(reservations) == 1:
+                for line in folio.reservation_id.reservation_line:
+                    for room in line.reserve:
+                        vals = {
+                            "room_id": room.id,
+                            "check_in": folio.checkin_date,
+                            "check_out": folio.checkout_date,
+                            "state": "assigned",
+                            "reservation_id": folio.reservation_id.id,
+                        }
+                        reservations.write(vals)
+        return res
+
+
+class HotelFolioLineExt(models.Model):
+
+    _inherit = "hotel.folio.line"
+
+    @api.onchange("checkin_date", "checkout_date")
+    def on_change_checkout(self):
```

Comment:
```suggestion
    def _onchange_checkout_dates(self):
```

make sure to change it at affected places as well

Reply:
Done